custom_components/localtuya/climate.py

```diff
@@ -0,0 +1,220 @@
+"""Platform to locally control Tuya-based climate devices."""
+import logging
+from functools import partial
+
+import voluptuous as vol
+
+from homeassistant.components.climate import (
+    ClimateEntity,
+    DOMAIN,
+    DEFAULT_MIN_TEMP,
+    DEFAULT_MAX_TEMP,
+)
+from homeassistant.components.climate.const import (
+    HVAC_MODE_AUTO,
+    # HVAC_MODE_COOL,
+    # HVAC_MODE_FAN_ONLY,
+    HVAC_MODE_HEAT,
+    HVAC_MODE_OFF,
+    SUPPORT_TARGET_TEMPERATURE,
+    SUPPORT_TARGET_TEMPERATURE_RANGE,
+    # SUPPORT_TARGET_HUMIDITY,
+    SUPPORT_FAN_MODE,
+    # SUPPORT_PRESET_MODE,
+    # SUPPORT_SWING_MODE,
+    # SUPPORT_AUX_HEAT,
+)
+from homeassistant.const import (
+    ATTR_TEMPERATURE,
+    CONF_TEMPERATURE_UNIT,
+    TEMP_CELSIUS,
+    TEMP_FAHRENHEIT,
+    PRECISION_WHOLE,
+    PRECISION_HALVES,
+    PRECISION_TENTHS,
+)
+
+from .const import (
+    CONF_TARGET_TEMPERATURE_DP,
+    CONF_CURRENT_TEMPERATURE_DP,
+    CONF_MAX_TEMP_DP,
+    CONF_MIN_TEMP_DP,
+    CONF_FAN_MODE_DP,
+    CONF_HVAC_MODE_DP,
+    CONF_TEMPERATURE_STEP,
+    CONF_PRECISION,
+)
+from .common import LocalTuyaEntity, async_setup_entry
+
+_LOGGER = logging.getLogger(__name__)
+
+TEMPERATURE_CELSIUS = "celsius"
+TEMPERATURE_FAHRENHEIT = "fahrenheit"
+
+
+def flow_schema(dps):
+    """Return schema used in config flow."""
+    return {
+        vol.Optional(CONF_TARGET_TEMPERATURE_DP): vol.In(dps),
+        vol.Optional(CONF_CURRENT_TEMPERATURE_DP): vol.In(dps),
+        vol.Optional(CONF_TEMPERATURE_STEP): vol.In(
+            [PRECISION_WHOLE, PRECISION_HALVES, PRECISION_TENTHS]
+        ),
+        vol.Optional(CONF_HVAC_MODE_DP): vol.In(dps),
+        vol.Optional(CONF_FAN_MODE_DP): vol.In(dps),
+        vol.Optional(CONF_MAX_TEMP_DP): vol.In(dps),
+        vol.Optional(CONF_MIN_TEMP_DP): vol.In(dps),
+        vol.Optional(CONF_PRECISION): vol.In(
+            [PRECISION_WHOLE, PRECISION_HALVES, PRECISION_TENTHS]
+        ),
+        vol.Optional(CONF_TEMPERATURE_UNIT): vol.In(
+            [TEMPERATURE_CELSIUS, TEMPERATURE_FAHRENHEIT]
+        ),
+    }
+
+
+class LocaltuyaClimate(LocalTuyaEntity, ClimateEntity):
+    """Tuya climate device."""
+
+    def __init__(
+        self,
+        device,
+        config_entry,
+        switchid,
+        **kwargs,
+    ):
+        """Initialize a new LocaltuyaClimate."""
+        super().__init__(device, config_entry, switchid, **kwargs)
+        self._state = None
+        self._target_temperature = None
+        self._current_temperature = None
+        self._hvac_mode = None
+        self._preset_mode = None
+        self._precision = PRECISION_TENTHS
+        if self.has_config(CONF_PRECISION):
```

Comment:
Maybe we can so something like this in these cases:

```python
self._precision = self._config.get(CONF_PRECISON, PRECISION_TENTHS)
```
It will save us some code and avoid double assignments.